src/sys/socket/sockopt.rs

```diff
@@ -1268,6 +1268,17 @@ sockopt_impl!(
     libc::SO_EXCLBIND,
     bool
 );
+#[cfg(target_os = "linux")]
```

Comment:
Hi there

> However, I'd like advice on how to resolve the latter.


If uClibc has that symbol, you can add it to the libc crate. Otherwise, we need to filter it out:

```suggestion
#[cfg(all(target_os = "linux", not(target_env = "uclibc")))]
```

Reply:
Added `tcp_info` to the uClibc symbols in libc: https://github.com/rust-lang/libc/commit/4a34937400c0606b908c7102782b485bee619f51

Reply:
That PR https://github.com/rust-lang/libc/pull/4347 added it to the main branch, Nix uses the [`libc-0.2`](https://github.com/rust-lang/libc/tree/libc-0.2) branch, so you need to ask the maintainer to backport it to this branch.

Once the backport PR gets merged, update the `Cargo.toml` file to use it from git:

https://github.com/nix-rust/nix/blob/ea012bec465a6d9341d8e4bd8b7c16100393c2aa/Cargo.toml#L31

```diff
- libc = { version = "0.2.171", features = ["extra_traits"] } 
+ libc = { git = "https://github.com/rust-lang/libc", rev = "the commit includes your libc PR", features = ... }
```

Reply:
Got it, thanks! I'll update this PR when the change is backported.

Reply:
I'm a bit confused by this comment. The commit was backported to the 0.2 branch: https://github.com/rust-lang/libc/commit/5a1f5f206beaf7845d40f3394758b468d69751f4. Why would pinning the version of the crate in the nix Cargo.toml file be the correct approach? I don't think that should get committed, right?

Reply:
Emm, I don't think I understand what you mean, could you please elaborate a bit?